### DISCUSSION TITLE
Error running DASimpleTFoam with different geometry

### DISCUSSION THREAD

**Discussion by sreekarrs** [mdolab/dafoam#421] [General]
Hello

I am trying to run DASimpleTFoam which has a U-Bend channel tutorial case with a different geometry but facing an error on Time = 1 in the optimization loop.

I am facing the following error:
```
Time = 1

U Initial residual: (nan nan nan)
U   Final residual: (nan nan nan)


--> FOAM FATAL ERROR: 
No coarse levels created, either matrix too small for GAMG or nCellsInCoarsestLevel too large.
    Either choose another solver of reduce nCellsInCoarsestLevel.

    From function Foam::GAMGSolver::GAMGSolver(const Foam::word&, const Foam::lduMatrix&, const Foam::FieldField<Foam::Field, double>&, const Foam::FieldField<Foam::Field, double>&, const lduInterfaceFieldPtrsList&, const Foam::dictionary&)
    in file matrices/lduMatrix/solvers/GAMG/GAMGSolver.C at line 270.

FOAM exiting
```

I have a sample geometry with a rectangular opening at the bottom center. I applied a constant temperature at the walls in the opening and have symmetry on the back face. This case has similar boundary conditions as the U-Bend tutorial. In the picture attached the red border is applied with a constant temperature.

![image](https://user-images.githubusercontent.com/26832867/230389509-90069344-d7f4-4b0e-82c7-a521bef0c1d7.png)

I would like to perform a similar simulation as U-Bend, that is reduce the pressure loss and increase heat flux by modifying the opening surface.
I attached the case set up in a tar file for reference. 
[dafoam_err.tar.gz](https://github.com/mdolab/dafoam/files/11171016/dafoam_err.tar.gz)

Thank you in advance,
Sree

**Comment by friedenhe** [comment]
Can you run your setup using native OpenFOAM solvers, such as simpleFoam to see if it works? Also, make sure you are not using the `empty` boundary condition; it is not supported in DAFoam. Check [this](https://dafoam.github.io/mydoc_get_started_faq.html#does-dafoam-support-optimization-for-pure-2d-problems).

**Reply by sreekarrs** [reply]
Thank you for the response

I ran simpleFoam with just fluid flow and also ran buoyantSimpleFoam with similar temperature boundary conditions. I have results from both simulations and the residuals are well below 1e-5 after ~600 iterations. I used the same case setup for DASimpleTFoam simulation, but I am getting the error.
The geometry is similar to the U-bend case; the front face is a wall and the other face is symmetry. I don't have any empty faces.

I could see from the log file that there are no results in checkMesh when I execute run script. But when I run checkMesh separately, it is showing all the parameters.

```
Starting time loop

Checking mesh quality for time = 0
    Overall domain bounding box (nan -nan nan) (nan -nan nan)
    Mesh has 3 geometric (non-empty/wedge) directions 3{1}
    Mesh has 3 solution (non-empty) directions 3{1}
    Boundary openness (-nan nan nan) OK.
    Max cell openness = nan OK.
    Max aspect ratio = nan OK.
    Minimum face area = nan. Maximum face area = nan.  Face area magnitudes OK.
    Min volume = nan. Max volume = nan.  Total volume = nan.  Cell volumes OK.
    Mesh non-orthogonality Max: -nan average: -nan
    Non-orthogonality check OK.
    Face pyramids OK.
    Max skewness = nan OK.
    Coupled point location match (average 0) OK.

Mesh OK.

Time = 1

U Initial residual: (nan nan nan)
U   Final residual: (nan nan nan)

```

Here is the complete file [logOpt.txt](https://github.com/mdolab/dafoam/files/11258842/logOpt.txt)

I also attached the case setup in the post above.
Thank you for your help
-- Sreekar

**Comment by friedenhe** [comment]
Your FFD did not fully contain the design surface. Change the FFD file.
